src/serialize.rs

```diff
@@ -570,14 +570,18 @@ impl Decodable for path::PathBuf {
         use std::os::unix::prelude::*;
         let bytes: Vec<u8> = try!(Decodable::decode(d));
         let s: OsString = OsStringExt::from_vec(bytes);
-        Ok(path::PathBuf::new(&s))
+        let mut p = path::PathBuf::new();
+        p.push(s);
```

Comment:
I'm a bit unsure about this part. I hate mut. Might it be better to use FromIterator here?